### PR TITLE
Fix PiecewiseChebyshev for constant functions

### PR DESCRIPTION
**Pull request by anedumla**
### Summary
There was an issue raised in [this](https://quantumcomputing.stackexchange.com/questions/16899/question-on-qiskit-circuit-library-piecewisechebyshev-qiskit-circuit-library) Stackexchange question in which the code was not replicating the expected result when given as the input function to be approximated a constant function.

### Details and comments
The error came from trying to approximate such function with `numpy.polynomial.chebyshev.interpolate`. The issue has been fixed, an error message detailing how to declare constant functions added and the documentation updated with a statement on how to declare constant functions.

### CODE REVIEW

File: releasenotes/notes/fix_pwchebysev_constant_fx-93e8a3d2880f68ac.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+fixes:
+  - |
+    There was a problem when the function to be approximated was constant.
```
Comment (woodsp-ibm) [inline releasenotes/notes/fix_pwchebysev_constant_fx-93e8a3d2880f68ac.yaml:4]: Would you provide a bit of context in the fix note here. When its published as a release note, where this is what is being read, it talks about constant function bug but there is no other context here to understand it better.

When you update this there is also a minor spelling error ` funcions.`

Comment (anedumla) [reply releasenotes/notes/fix_pwchebysev_constant_fx-93e8a3d2880f68ac.yaml:4]: I've updated the release note and fixed the typo.

**Comment by woodsp-ibm** [comment]
Since I guess this is not an aspect that was covered by testing is there a simple unit test that can be added to ensure the correctness and that this stays working?

**Comment by anedumla** [comment]
> 
> 
> Since I guess this is not an aspect that was covered by testing is there a simple unit test that can be added to ensure the correctness and that this stays working?

There is a test for a constant function but not declaring it as lambda - this would raise an error that is now getting caught.

File: qiskit/circuit/library/arithmetic/piecewise_chebyshev.py [hunk 2]
```diff
@@ -64,7 +63,8 @@ def __init__(
     ) -> None:
         r"""
         Args:
-            f_x: the function to be approximated.
+            f_x: the function to be approximated. Constant functions should be specified
```
Comment (Cryoris) [inline qiskit/circuit/library/arithmetic/piecewise_chebyshev.py:66]: So for a constant function I would call
```
PiecewiseChebychev(0.5, degree=3, etc.)
```
? If yes, could you add floats in the typehint for `f_x`? 🙂

Comment (anedumla) [reply qiskit/circuit/library/arithmetic/piecewise_chebyshev.py:66]: exactly, added float to the typehint

**Review by Cryoris — APPROVED** [review]
LGTM thanks for the fix 👍🏻

**Pull request merged**